### PR TITLE
Update request body schemas

### PR DESCRIPTION
**Pull request by waynesun09**
Add new TextBody and DictBody schema for request body, use Union for
support multiple schemas.

Update apis with the schemas, this change api behavior with 'body'
key is required in request body.

Also update examples for request body.

Signed-off-by: Wayne Sun <gsun@redhat.com>

### CODE REVIEW

File: app/api/endpoints/chat.py [hunk 1]
```diff
@@ -21,8 +21,11 @@ async def send_message(
     template_name: str = Query(
         "chat_default",
         description="The template name without suffix, e.g. chat_default, default to 'chat_default' for gchat"),
```
Comment (abraverm) [inline app/api/endpoints/chat.py:23]: could you add a the link https://github.com/waynesun09/notify-service/blob/main/app/templates/build/chat_default.jinja to 'chat_default'?

File: app/api/endpoints/chat.py [hunk 2]
```diff
@@ -21,8 +21,11 @@ async def send_message(
...
-    environment: Dict[str, Any] = Body(
-        {"body": ""},
+    environment: Union[schemas.DictBody, schemas.TxtBody] = Body(
+        ...,
+        example={
+            "body": "SAMPLE MESSAGE USE THE DEFAULT CHAT TEMPLATE, REPLACE IT OR USE DIFFERENT TEMPLATE AND CONTENT."
```
Comment (abraverm) [inline app/api/endpoints/chat.py:27]: I think "Sample message" is enough

File: app/api/endpoints/chat.py [hunk 3]
```diff
@@ -21,8 +21,11 @@ async def send_message(
...
+        },
         description="The body values for parse with the template"),
```
Comment (abraverm) [inline app/api/endpoints/chat.py:29]: I would mention Jinja here and 'chat_default' with a link to https://github.com/waynesun09/notify-service/blob/main/app/templates/build/chat_default.jinja

**Review by abraverm — COMMENTED** [review]
I remember seeing in the generated doc the return value type, maybe you can add there an example too with how "Sample message" parsed with "chat_default" template to:
```
{
  "text": "Sample Message"
}
```

File: app/api/endpoints/chat.py (hunk 1 above)
Comment (waynesun09) [reply app/api/endpoints/chat.py:23]: sure

File: app/api/endpoints/chat.py (hunk 2 above)
Comment (waynesun09) [reply app/api/endpoints/chat.py:27]: OK

**Comment by waynesun09** [comment]
@abraverm updated, pls check, thanks

**Review by abraverm — APPROVED** [review]

**Pull request merged**